apps/andi/test/unit/andi/event/event_handler_test.exs

```diff
@@ -17,9 +16,36 @@ defmodule Andi.Event.EventHandlerTest do
   alias Andi.Services.OrgStore
   alias Andi.InputSchemas.Ingestions
   alias Andi.Services.IngestionStore
+  alias SmartCity.DatasetAccessGroupRelation
 
   @instance_name Andi.instance_name()
 
+  test "should successfully handle a dataset_access_group_associate() event" do
+    dataset_id = UUID.uuid4()
+    access_group_id = UUID.uuid4()
+    {:ok, assoc} = DatasetAccessGroupRelation.new(%{dataset_id: dataset_id, access_group_id: access_group_id})
+    allow(Andi.InputSchemas.Datasets.Dataset.associate_with_access_group(access_group_id, dataset_id), return: :ok)
+    expect(TelemetryEvent.add_event_metrics(any(), [:events_handled]), return: :ok)
+
+    event = Brook.Event.new(type: dataset_access_group_associate(), data: assoc, author: :author)
+    EventHandler.handle_event(event)
+
+    assert_called(Andi.InputSchemas.Datasets.Dataset.associate_with_access_group(access_group_id, dataset_id))
+  end
```

Comment:
Do we want the assertion here to confirm that the association ended up in AccessGroup table? Or was it a deliberate choice to not assert side effects.

I see how by asserting that `Dataset.associate_with...` was called, it's testing that the event handler fulfilled it's responsibility.

Previously @LtChae advised that we might as well test the full effect to get some more coverage, but I like how you've done it here, since it's a Unit test. 

I feel like unit tests should only test the specific piece they're testing, and not their effects elsewhere. 

(ex: if we asserted it ended up in the DB, and we broke the Datasets inputschema, the event handler test would break, which doesn't seem right)

---

So, not really requesting a change here. Just curious of your thought process on what to assert for this was. Would like to align the way I think about it for future tickets.